examples/bert/run_pretraining.py

```diff
@@ -291,6 +301,38 @@ def train_step(self, data):
         self.nsp_accuracy.update_state(data["next_sentence_labels"], nsp_preds)
         return {m.name: m.result() for m in self.metrics}
 
+class LinearDecayWithWarmup(keras.optimizers.schedules.LearningRateSchedule):
+    """
+    Implements a learning rate schedule for BERT.
+
+    This schedule implements a linear warmup for the first `num_warmup_steps` 
+    and for the steps post that, till the `num_train_steps` it will implement 
+    a linear decay.
+    """
+    
+    def __init__(self,
+            learning_rate,
+            num_warmup_steps,
+            num_train_steps
+        ):
+        self.threshold_learning_rate = learning_rate
```

Comment:
Keep the same name as argument...

self.learning_rate = learning_rate

Reply:
Okay, I'll change it